src/System.Net.Http/tests/FunctionalTests/HuffmanDecoder.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Functional.Tests
+{
+    // Stolen from product code
```

Comment:
Could we just include the relevant product file, separating the relevant members out if necessary?

Reply:
Maybe, but it gets messy. It didn't seem worth it, so I just copied the code. How important do you think it is?

Reply:
It's not critical.  It's just easily going to get out of date as the product code evolves, it's extra code to maintain, etc.